packages/io/x3d-serializer/src/index.js

```diff
@@ -52,39 +60,43 @@ const mimeType = 'model/x3d+xml'
  */
 const serialize = (options, ...objects) => {
   const defaults = {
-    unit: 'millimeter', // millimeter, inch, feet, meter or micrometer
     color: [0, 0, 1, 1.0], // default colorRGBA specification
     decimals: 1000,
+    metadata: true,
+    unit: 'millimeter', // millimeter, inch, feet, meter or micrometer
     statusCallback: null
   }
   options = Object.assign({}, defaults, options)
 
   objects = flatten(objects)
 
-  // convert only 3D geometries
-  let objects3d = objects.filter((object) => geometries.geom3.isA(object))
-
-  if (objects3d.length === 0) throw new Error('only 3D geometries can be serialized to X3D')
-  if (objects.length !== objects3d.length) console.warn('some objects could not be serialized to X3D')
+  objects = objects.filter((object) => geom3.isA(object) || geom2.isA(object) || path2.isA(object))
 
-  // convert to triangles
-  objects3d = toArray(modifiers.generalize({ snap: true, triangulate: true }, objects3d))
+  if (objects.length === 0) throw new Error('only JSCAD geometries can be serialized to X3D')
```

Comment:
What does it mean "only JSCAD geometries"? It seems like all geometries are now supported. Maybe this message would be more accurately "geometry must be non-empty to serialize to X3D"?